### ISSUE TITLE
Open .wav file in audio_operation.py: line 162 casts out error

### ISSUE THREAD

**Issue by zeng-yifei** [miaomiaomiao-LJY/ppg_tacotron#5]
when using 'open' function to read .wav file like this

`open('/content/data/dataset/arctic/bdl/arctic_a0001.wav',encoding='utf-8').read().splitlines()`

Error occurs and indicates a decoding failure:

>'utf-8' codec can't decode byte 0x86 in position 4: invalid start byte

Is that due to different python version? I'm wondering why this error happens.

**Comment by LJY-M** [comment]
> when using 'open' function to read .wav file like this
> 
> `open('/content/data/dataset/arctic/bdl/arctic_a0001.wav',encoding='utf-8').read().splitlines()`
> 
> Error occurs and indicates a decoding failure:
> 
> > 'utf-8' codec can't decode byte 0x86 in position 4: invalid start byte
> 
> Is that due to different python version? I'm wondering why this error happens.

Hello, this problem has nothing to do with the python version. 
The function of this code is to open the phoneme label file and record the phoneme and its duration.

`for line in open(phn_file, encoding='utf-8').read().splitlines():`

 In this code, the variable **phn_file** represents the phoneme tag file (.pth) instead of the audio file (.wav), and you can refer to the Timit dataset to see the data format.

**Comment by zeng-yifei** [comment]
Oh, I made a careless mistake. Thank you very much for reminding!

**Comment by zeng-yifei** [comment]
Plus, do the TIMIT content contains file such as this?
`file.TXT, file.WAV, file.WAV.wav, file.WRD, file.PHN`
I didnt download from official website. So I'm hesitating whether the format, especially the uppercase, is right or not.

**Comment by LJY-M** [comment]
> Plus, do the TIMIT content contains file such as this? `file.TXT, file.WAV, file.WAV.wav, file.WRD, file.PNH` I didnt download from official website. So I'm hesitating whether the format, especially the uppercase, is right or not.

The original TIMIT dataset does not have a file like "**file.WAV.wav**", other file names are correct.
However, there is a problem with the original "**file.WAV**" file format and it needs to be further processed to match the normal "**file.wav**" file, which may have already been processed for the dataset you downloaded.

**Comment by zeng-yifei** [comment]
So, does the path`net1_dataset = "../data/dataset/TIMIT/TRAIN/*/*/*.wav"` in hparam.py indicate the **processed** .wav file series? In another word,  should the files' format be like this after processing?
`file.TXT, file.wav, file.WRD, file.PHN`

**Comment by LJY-M** [comment]
> So, does the path`net1_dataset = "../data/dataset/TIMIT/TRAIN/*/*/*.wav"` in hparam.py indicate the **processed** .wav file series? In another word, should the files' format be like this after processing? `file.TXT, file.wav, file.WRD, file.PHN`

Yes, the variable `net1_dataset` points to the processed TIMIT dataset.
In this project, the processed audio file is named `file_test.wav` or `file_train.wav`, depending on the directory it belongs to. Other files are not processed.

**Comment by zeng-yifei** [comment]
I finally fixed this problem by setting the *.WAV.wav into *_train.wav .Thank you very much for your guidance!

**Comment by geni120** [comment]
> I finally fixed this problem by setting the *.WAV.wav into *_train.wav .Thank you very much for your guidance!

I am also getting UTF-8 error. Can you please give more details on how you are able to solve UTF-8 issue?

**Comment by WildStrom** [comment]
got the same problem, how to solve it? change the file name?

**Comment by WildStrom** [comment]
solved the problem, by change the code as fllows:
原来的：phn_file = wav_file.replace("_train.wav", ".PHN").replace("_test.wav", ".PHN")
修改为：phn_file = wav_file.replace(".WAV", ".PHN").replace(".WAV", ".PHN")

**Comment by LJY-M** [comment]
> solved the problem, by change the code as fllows: 原来的：phn_file = wav_file.replace("_train.wav", ".PHN").replace("_test.wav", ".PHN") 修改为：phn_file = wav_file.replace(".WAV", ".PHN").replace(".WAV", ".PHN")

It looks like the problem has been solved. Congratulations.
